app/src/main/java/http_server/ResponseHandler.java

```diff
@@ -9,6 +11,12 @@ public Response buildResponse( String status, String responseBody) {
     return response;
   }
 
+  public Response buildResourceResponse(String method, String options, String status, String responseBody) {
+    response.setResponseStatus(status);
+    response.setResponseHeaders(method, options);
+    response.setResponseBody(responseBody);
+    return response;
```

Comment:
This is Soooo close to the builder pattern, another pattern worth looking into (for future).

Reply:
Awesome! I remember you mentioning this pattern in our pairing session yesterday. I'll check it out!